tests/buffer/buffer_api_common.h

```diff
@@ -52,12 +52,33 @@ inline void precalculate<2>(cl::sycl::range<2> &rangeIn,
 template <>
 inline void precalculate<3>(cl::sycl::range<3> &rangeIn,
                             cl::sycl::range<3> &rangeOut, size_t &elementsCount,
-                            unsigned elementsIn, unsigned elementsOut) {
+                            size_t elementsIn, size_t elementsOut) {
   rangeIn = cl::sycl::range<3>(elementsIn, elementsIn, elementsIn);
   rangeOut = cl::sycl::range<3>(elementsOut, elementsIn, elementsIn);
   elementsCount = (elementsOut * elementsIn * elementsIn);
 }
 
+/*!
+@brief used to calculate the element count of a multi-dimensional buffer
+*/
+template <int dims>
+inline size_t calc_elem_count(size_t numElemsPerDim);
+
+template <>
+inline size_t calc_elem_count<1>(size_t numElemsPerDim) {
+  return numElemsPerDim;
+}
+
+template <>
+inline size_t calc_elem_count<2>(size_t numElemsPerDim) {
+  return numElemsPerDim * numElemsPerDim;
+}
+
+template <>
+inline size_t calc_elem_count<3>(size_t numElemsPerDim) {
+  return numElemsPerDim * numElemsPerDim * numElemsPerDim;
+}
+
```

Comment:
```suggestion
template <int dims>
inline size_t calc_elem_count(size_t numElemsPerDim) {
  return std::pow(numElemsPerDim, dims);
}
```
is painful enough.
I wonder whether we actually need a function template instead of a function...

Reply:
OK remove this...